concepts/function-arguments/about.md

```diff
@@ -1,2 +1,245 @@
+Positional arguments can optionally be passed by using their parameter name.
+
+Following is an example of positional arguments being passed by position and by their parameter name:
+
+```python
+>>> def concat(greeting, name):
+...         return f"{greeting}{name}"
+... 
+# Passing data to the function by position.
+>>> print(concat("Hello, ", "Bob"))
+
+Hello, Bob
+...
+# Passing data to the function using the parameter name.
+>>> print(concat(name="Bob", greeting="Hello, "))
+
+Hello, Bob
+
+```
+
+The first call to `concat` passes the arguments by position.
+The second call to `concat` passes the arguments by name, allowing their positions to be changed.
+
+Note that positional arguments cannot follow keyword arguments.
+
+This
+
+```python
+>>> print(concat(greeting="Hello, ", "Bob"))
+```
+
+results in this error:
+
+```
+SyntaxError: positional argument follows keyword argument
+```
+
+Requiring positional-only arguments for function calls can be done through the use of the `/` operator in the parameter list:
+
+Following is an example of positional-only arguments:
+
+```python
+# Parameters showing a position-only operator. 
+>>> def concat(greeting, name, /):
+...         return f"{greeting}{name}"
+
+... 
+>>> print(concat("Hello, ", "Bob"))
+Hello, Bob
+...
+# Call to the function using keyword arguments.
+>>> print(concat(name="Bob", greeting="Hello, "))
+Traceback (most recent call last):
+    print(concat(name="Bob", greeting="Hello, "))
+TypeError: concat() got some positional-only arguments passed as keyword arguments: 'greeting, name'
+
+
+```
+
+## Keyword Arguments
+
+Keyword arguments use the parameter name when calling a function.
+Keyword arguments can optionally be referred to by position.
+
+Following is an example of keyword arguments being referred to by their parameter name and by position:
+
+```python
+>>> def concat(greeting, name):
+...         return f"{greeting}{name}"
+... 
+# Function call using parameter names as argument keywords.
+>>> print(concat(name="Bob", greeting="Hello, "))
+Hello, Bob
+...
+# Function call with positional data as arguments.
+>>> print(concat("Hello, ", "Bob"))
+Hello, Bob
+>>> print(concat())
+Hello, you
+
+```
+
+Requiring keyword-only arguments for function calls can be done through the use of the `*` operator in the parameter list:
+
+Following is an example of keyword-only arguments:
+
+```python
+# Function definition requiring keyword-only arguments.
+>>> def concat(*, greeting, name):
+...         return f"{greeting}{name}"
+... 
+# Keyword arguments can be in an arbitrary order.
+>>> print(concat(name="Bob", greeting="Hello, "))
+Hello, Bob
+...
+# Calling the function with positional data raises an error.
+>>> print(concat("Hello, ", "Bob"))
+Traceback (most recent call last):
+    print(concat("Hello, ", "Bob"))
+TypeError: concat() takes 0 positional arguments but 2 were given
+
+
+```
+

```

Comment:
```suggestion

## Default Argument Values

Default values for one or more arguments can be supplied in the parameter list.
This allows the function to be called with _fewer_ arguments if needed.
Default values can be overridden by calling the function with a new argument value, overriding the default:

```python
# Function with default argument values.
>>> def concat(greeting, name="you", punctuation="!"):
...         return f"{greeting}, {name}{punctuation}"
... 
>>> print(concat("Hello"))
Hello, you!

# Overriding the default values
>>> print(concat("Hello", name="Bob", punctuation="."))
Hello, Bob.
```
```